src/transformers/models/gptsan_japanese/modeling_gptsan_japanese.py

```diff
@@ -0,0 +1,1410 @@
+        tokenizer = AutoTokenizer.from_pretrained("Tanrei/GPTSAN-japanese")
+        x_token = tokenizer.encode(
+            "", prefix_text="武田信玄は、<|inputmask|>時代ファンならぜひ押さえ<|inputmask|>きたい名将の一人。", return_tensors="pt"
+        ).to(device)
+        trainer_utils.set_seed(30)
+        out_lm_token = model.generate(x_token, max_new_tokens=50)
+        out_mlm_token = model(x_token).logits.argmax(axis=-1)
+        tokenizer.decode(out_mlm_token[0])
+        "武田信玄は、戦国時代ファンならぜひ押さえておきたい名将の一人。"
+        tokenizer.decode(out_lm_token[0][x_token.shape[1] :])
+        "武田氏の三代に渡った武田家のひとり\n甲斐市に住む、日本史上最大の戦国大名。..."
+        ```"""
+        SEP_TOKEN = self.config.pad_token_id
+        NOT_TOKEN = self.config.unk_token_id
+        SEG_TOKEN = self.config.separator_token_id
+        use_cache = use_cache or self.config.use_cache
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+        model_return_dict = True
+        num_precontext = None
+        if input_ids is not None:
+            num_batch = input_ids.shape[0]
+            num_precontext = torch.zeros([num_batch]).int().to(input_ids.device)
+            where_separators = torch.where(input_ids == SEG_TOKEN)
+            num_precontext[where_separators[0]] += where_separators[1]
+            num_precontext = num_precontext.unsqueeze(1)
+
+        outputs = self.model(
+            input_ids,
+            attention_mask,
+            spout,
+            past_key_values,
+            head_mask,
+            use_cache,
+            inputs_embeds,
+            decoder_inputs_embeds,
+            output_attentions,
+            output_hidden_states,
+            model_return_dict,
+            output_router_logits,
+            num_precontext,
+        )
+
+        lm_logits = self.lm_head(outputs[0])
+        if lm_logits.shape[-1] == self.final_logits_bias.shape[-1]:
+            lm_logits = lm_logits + self.final_logits_bias
+
+        loss = None
+        z_loss = None
+        router_probs = None
+        aux_loss = None
+        if labels is not None:
+            loss_fct = nn.CrossEntropyLoss(ignore_index=-100)
+
+            if output_router_logits:
+                # Compute the router loss (z_loss + auxiliary loss) for each router in the encoder and decoder
+                router_logits, expert_indexes = self._unpack_router_logits(outputs.router_probs)
+                z_loss = router_z_loss_func(router_logits)
+                router_probs = nn.Softmax(dim=-1)(router_logits)
+                aux_loss = load_balancing_loss_func(router_probs, expert_indexes)
+
+            loss = loss_fct(lm_logits.view(-1, lm_logits.size(-1)), labels.view(-1))
+
+        # Ignore the two tokens when generating
+        if SEP_TOKEN < lm_logits.shape[2]:
+            lm_logits[:, :, SEP_TOKEN] = -1e10
+        if NOT_TOKEN < lm_logits.shape[2]:
+            lm_logits[:, :, NOT_TOKEN] = -1e10
```

Comment:
Things that modify the logits like this can be handled by the `logits_processor`. Pretty sure we have some logits processor that can prevent tokens from being generated ( `SuppressTokensLogitsProcessor`) for example.

Reply:
I have found this not really necessary.
I think only I need is add bad_words_ids to the model card's generation_config.json.

Reply:
Yes exactly!